apifile.go

```diff
@@ -0,0 +1,254 @@
+package httpapi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-ipfs-files"
+	"github.com/ipfs/interface-go-ipfs-core"
+)
+
+const forwardSeekLimit = 1 << 14 //16k
+
+func (api *UnixfsAPI) Get(ctx context.Context, p iface.Path) (files.Node, error) {
+	if p.Mutable() { // use resolved path in case we are dealing with IPNS / MFS
+		var err error
+		p, err = api.core().ResolvePath(ctx, p)
```

Comment:
It does, but we will be calling multiple enpoints with this path so we want to make sure we use the same resolved path for all (in case ipns name changes we'll keep reading valid data). It should also be faster.

Reply:
Got it.